pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/elastic/export/tabular/TabularParticipantExporter.java

```diff
@@ -42,29 +50,74 @@ public static TabularParticipantExporter getExporter(List<ModuleExportConfig> mo
         return new TsvParticipantExporter(moduleConfigs, participantValueMaps, fileFormat);
     }
 
+    /** returns a filename such that alphabetical sorting will also put them in chronological order */
     protected static String getExportFilename(String suffix) {
         LocalDate date = LocalDate.now();
         DateTimeFormatter formatter = DateTimeFormatter.ofPattern(FILE_DATE_FORMAT);
         String exportFileName = String.format("Participant-%s.%s", date.format(formatter), suffix);
         return exportFileName;
     }
 
+    /**
+     * Gets all the column names that need to be rendered for a given config, including repeats, details, and/or
+     *  splitting options across multiple column.  The column name is based off the stable name of the associated item
+     *  and needs to be globally unique
+     */
+
     public static List<String> getAllColumnNames(FilterExportConfig filterConfig, int formRepeatNum, int questionRepeatNum) {
+        List<String> allColNames = new ArrayList<>();
+        List<String> optionStableIds = Collections.singletonList(null);
+        if (filterConfig.isSplitOptionsIntoColumns()) {
+            optionStableIds = filterConfig.getOptions().stream().map(opt -> (String) opt.get(ESObjectConstants.OPTION_STABLE_ID))
+                    .collect(Collectors.toList());
+        }
+        allColNames = optionStableIds.stream().map(opt -> {
+                    List<String> colNames = new ArrayList<>();
+                    if (filterConfig.getChildConfigs() != null) {
+                        colNames.addAll(filterConfig.getChildConfigs().stream().map(childConfig -> getColumnName(filterConfig, formRepeatNum,
+                                questionRepeatNum, opt,null, childConfig)).collect(Collectors.toList()));
+                    } else {
+                        colNames.add(getColumnName(filterConfig, formRepeatNum, questionRepeatNum, opt,null, null));
+                        if (filterConfig.isHasDetails()) {
+                            colNames.add(getColumnName(filterConfig, formRepeatNum, questionRepeatNum, opt,"DETAIL", null));
+                        }
+                    }
+                    return colNames;
+                }).flatMap(Collection::stream).collect(Collectors.toList());
+
+
+        return allColNames;
+    }
+
+
+    /**
+     * Gets all the column texts to be rendered.  This is a human-readable counterpart to the column name.  e.g.
+     *  while the column name might be "MEDICIAL_HISTORY_2.ANALYSIS_TYPE" the text might be "Type of sample analysis"
+     */
+    public List<String> getAllColumnTexts(FilterExportConfig filterConfig) {
+        List<String> allColTexts = new ArrayList<>();
+        List<Map<String, Object>> splitOptions = Collections.singletonList(null);
```

Comment:
the same about `Collections.emtpyList();`

Reply:
same, will see if I can refactor the surroundings